src/java.base/share/classes/javax/security/auth/Subject.java

```diff
@@ -319,6 +329,181 @@ public Subject run() {
         });
     }
 
+
+    private static enum SubjectStorage {
+        ACC, THREAD_LOCAL, SCOPE_LOCAL
+    };
+
+    private static final SubjectStorage SUBJECT_STORAGE
+            = SharedSecrets.getJavaLangAccess().systemAllowSecurityManager()
+                    ? SubjectStorage.ACC
+                    : SubjectStorage.THREAD_LOCAL;
+
+//    private static final ScopeLocal<Subject> SUBJECT_SCOPE
+//            = ScopeLocal.inheritableForType(Subject.class);
+
+    private static final InheritableThreadLocal<Subject> SUBJECT_THREAD_LOCAL =
+            new InheritableThreadLocal<Subject>() {
+                @Override protected Subject initialValue() {
+                    return null;
+                }
+            };
+
+    /**
+     * Return the current installed subject.
+     * <p>
+     * The current installed subject is installed by the {@link #call}
+     * or {@link #run} method. When either {@code call(subject, action)} or
+     * {@code run(subject, action)} is called, {@code action} is executed
```

Comment:
Should this be `callAs` or `doAs`?

Reply:
Correct, it should be `callAs` and `getAs`.